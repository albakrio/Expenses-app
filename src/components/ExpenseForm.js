import React, { useState } from 'react';
import moment from 'moment';
import 'react-dates/initialize';
import { SingleDatePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';

const ExpenseForm = () => {
	const [description, setDescription] = useState('');
	const [note, setNote] = useState('');
	const [amount, setAmount] = useState('');
	const [createdAt, setCreatedAt] = useState(moment());
	const [focusedCalendar, setFocusedCalendar] = useState(false);
	const [error, setError] = useState('')

	const onDescriptionChange = (e) => {
		const newDescription = e.target.value;

		setDescription(newDescription);
	};

	const onNoteChange = (e) => {
		const newNote = e.target.value;
		setNote(newNote);
	};

	const onAmountChange = (e) => {
		const newAmount = e.target.value;
		if (!newAmount || newAmount.match(/^\d{1,}(\.\d{0,2})?$/)) {
			return setAmount(newAmount);
		}
	};

	const onDateChange = (createdAt) => {
		if (createdAt) {
			setCreatedAt(createdAt);
		}
	};
	const onFocusCalendar = ({ focused }) => {
		setFocusedCalendar(focused);
	};

	const onSubmitHandler = (e)=>{
		e.preventDefault()
		if(!description || !amount){
			setError('Please provide description and amount!')
		} else {
			setError('')
			console.log('submitted')

		}
	}

	return (
		<div>
		  {error && <p>{error}</p>}
			<form onSubmit={onSubmitHandler}>
				<input
					type='text'
					placeholder='Description'
					autoFocus
					value={description}
					onChange={onDescriptionChange}
				/>
				<input
					type='text'
					placeholder='Amount'
					value={amount}
					onChange={onAmountChange}
				/>
				<SingleDatePicker
					date={createdAt}
					onDateChange={onDateChange}
					focused={focusedCalendar}
					onFocusChange={onFocusCalendar}
					numberOfMonths={1}
					isOutsideRange={() => false}
				/>
				<textarea
					placeholder='write a note (optional)'
					value={note}
					onChange={onNoteChange}
				></textarea>
				<button> Add Expense </button>
			</form>
		</div>
	);
};

export default ExpenseForm;
