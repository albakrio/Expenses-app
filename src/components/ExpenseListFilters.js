import React, { useState } from 'react';
import { connect } from 'react-redux';
import {
	setTextFilter,
	sortByDate,
	sortByAmount,
	setStartDate,
	setEndDate
} from '../actions/filters';
import moment from 'moment';
import 'react-dates/initialize';

import { DateRangePicker } from 'react-dates';


const ExpenseListFilters = (props) => {
	//console.log(props);
	const [calendarFocused, setCalendarFocused] = useState(null);
	//console.log(calendarFocused);

	const onDatesChange = ({ startDate, endDate }) => {
		props.dispatch(setStartDate(startDate));

		props.dispatch(setEndDate(endDate));
	};

	const onFocusChange = ({ focusedInput }) => {
		setCalendarFocused(focusedInput);
	};
	return (
		<div>
			<input
				type='text'
				value={props.filters.text}
				onChange={(e) => {
					props.dispatch(setTextFilter(e.target.value));
				}}
			/>
			<select
				value={props.filters.sortBy}
				onChange={(e) => {
					if (e.target.value === 'date') {
						props.dispatch(sortByDate());
					} else if (e.target.value === 'amount') {
						props.dispatch(sortByAmount());
					}
				}}
			>
				<option value='date'>Date</option>
				<option value='amount'>Amount</option>
			</select>

			<DateRangePicker
				startDate={props.filters.startDate} // momentPropTypes.momentObj or null,
				startDateId='start' // PropTypes.string.isRequired,
				endDate={props.filters.endDate} // momentPropTypes.momentObj or null,
				endDateId='end' // PropTypes.string.isRequired,
				onDatesChange={onDatesChange}
				focusedInput={calendarFocused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
				onFocusChange={onFocusChange} // PropTypes.func.isRequired,
			/>
		</div>
	);
};

const mapStateToProps = (state) => {
	return {
		filters: state.filters,
		expenses: state.expenses
	};
};

export default connect(mapStateToProps)(ExpenseListFilters);
