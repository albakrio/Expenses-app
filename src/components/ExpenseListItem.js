import React from 'react';
import {connect} from 'react-redux'
import { removeExpense } from '../actions/expenses';

const ExpenseListItem = ({amount,createdAt, description, dispatch, id}) => {
	
	return (
		<div>
			<h1>{description}</h1>
			<p>
				{amount} - {createdAt}
			</p>
			<button onClick={()=>{dispatch(removeExpense({id}))}}>Delete</button>
		</div>
	);
};

export default connect()(ExpenseListItem);
