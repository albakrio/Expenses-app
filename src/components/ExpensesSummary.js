import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expensesTotal';
import { Link } from 'react-router-dom';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
	const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
	const formattedExpenseTotal = numeral(expensesTotal / 1000).format('$0,0.00');
	return (
		<div className='page-header'>
			<div className="content-container">
				<h1 className="page-header__title">
					Viewing <span>{expenseCount}</span> {expenseWord}. Totalling <span>{formattedExpenseTotal}</span>
				</h1>
				<div className="page-header__actions">
				  <Link className="button" to="/create">New Expense</Link>
				</div>
			</div>
		</div>
	);
};

const mapStateToProps = (state) => {
	const visibleExpenses = selectExpenses(state.expenses, state.filters);

	return {
		expenseCount: visibleExpenses.length,
		expensesTotal: selectExpensesTotal(visibleExpenses)
	};
};

export default connect(mapStateToProps)(ExpensesSummary);
