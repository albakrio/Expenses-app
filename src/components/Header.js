import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => (
	<header>
		<h1>Expensify</h1>
		<Link to='/' exact className='is-active'>
			Dashboard
		</Link>
		<Link to='/create' className='is-active'>
			Create Expense
		</Link>
		<Link to='/help' className='is-active'>
			Help
		</Link>
	</header>
);

export default Header;
