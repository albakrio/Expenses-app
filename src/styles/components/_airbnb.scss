.SingleDatePicker {
  border: 1px solid #cacccd;
}

.SingleDatePickerInput__withBorder {
  margin: 1px;
}

.SingleDatePickerInput__withBorder {
  margin: 2px;
  border: none;
}

.SingleDatePickerInput {
  width: 99%;

  .DateInput {
    width: 100%;
  }
}