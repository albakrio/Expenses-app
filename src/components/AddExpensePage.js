import React from 'react';
import ExpenseForm from './ExpenseForm';
import {connect} from 'react-redux'
import {addExpense} from '../actions/expenses'

const AddExpensePage = (props) => (
	<div>
		<h1>Create an Expense !</h1>
		<ExpenseForm
			onSubmitResult={(expense) => {
        props.dispatch(addExpense(expense))
        props.history.push('/')
      }}
      
		/>
	</div>
);

export default connect()(AddExpensePage);
