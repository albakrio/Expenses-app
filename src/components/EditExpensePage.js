import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense } from '../actions/expenses';
import { startRemoveExpense } from '../actions/expenses';

const EditExpensePage = (props) => {
	console.log(props);
	return (
		<div>
			<ExpenseForm
				expense={props.expense}
				onSubmitResult={(expense) => {
					props.dispatch(startEditExpense(props.match.params.id, expense));
					props.history.push('/');
				}}
			/>

			<button
				onClick={() => {
          props.dispatch(startRemoveExpense({ id: props.expense.id }));
          props.history.push('/');
          
				}}
			>
				Delete
			</button>
		</div>
	);
};

const mapStateToProps = (state, props) => {
	return {
		expense: state.expenses.find(
			(expense) => expense.id === props.match.params.id
		)
	};
};

// const mapDispatchToProps = (dispatch,props)=>({
// 	editExpense: (id,expense)=> dispatch(editExpense(id,expense)),
//   startRemoveExpense: (data) => dispatch(startRemoveExpense(data))

// })

export default connect(mapStateToProps)(EditExpensePage);
