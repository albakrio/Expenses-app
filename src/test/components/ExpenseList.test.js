import React from 'react';
import { ExpenseList } from '../../components/ExpenseList';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';

test('should return ExpenseList component with expenses', () => {
	const wrapper = shallow(<ExpenseList expenses={expenses} />);
	expect(wrapper).toMatchSnapshot();
});

test('should return ExpenseList component with empty message', () => {
	const wrapper = shallow(<ExpenseList expenses={[]} />);
	expect(wrapper).toMatchSnapshot();
});
