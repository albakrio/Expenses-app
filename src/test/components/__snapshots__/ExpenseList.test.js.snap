// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return ExpenseList component with empty message 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;

exports[`should return ExpenseList component with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={10095}
    createdAt={0}
    description="rent"
    id={1}
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={100}
    createdAt={-345600000}
    description="Credit Card"
    id={2}
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={200}
    createdAt={345600000}
    description="bill"
    id={3}
    key="3"
    note=""
  />
</div>
`;
